src/nvim/os/env.c

```diff
@@ -557,9 +553,6 @@ void home_replace(buf_T *buf, char_u *src, char_u *dst, int dstlen, bool one)
     dirlen = STRLEN(homedir);
 
   homedir_env_orig = homedir_env = (char_u *)os_getenv("HOME");
-  /* Empty is the same as not set. */
-  if (homedir_env != NULL && *homedir_env == NUL)
-    homedir_env = NULL;
```

Comment:
Logic isn't entirely the same. Before, `homedir_env_orig` is later used to decided whether to free `homedir_env`. I haven't completely understood the logic, but it might be worth a closer look, or maybe `homedir_env_orig` can be eliminated.